metricbeat/beater/glob_change_detector.go

```diff
@@ -0,0 +1,79 @@
+package beater
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/mitchellh/hashstructure"
+)
+
+type GlobChangeDetector struct {
+	glob     string
+	lastScan time.Time
+	lastHash uint64
+}
+
+func NewGlobalChangeDetector(glob string) *GlobChangeDetector {
+	return &GlobChangeDetector{
+		lastScan: time.Time{},
+		lastHash: 0,
+		glob:     glob,
+	}
+}
+
+// Scan scans all files matched by the glob and checks if the number of files or the modtime of the files changed
+// It returns the list of files, a boolean if anything in the glob changed and potential errors.
+// To detect changes not only mod time is compared but also the hash of the files list. This is required to
+// also detect files which were removed.
+// The modtime is compared based on second as normally mod-time is in seconds. If it is unclear if something changed
+// the method will return true for the changes. It is strongly recommend to call scan not more frequent then 1s.
+func (gcd *GlobChangeDetector) Scan() ([]string, bool, error) {
+
+	globList, err := filepath.Glob(gcd.glob)
+	if err != nil {
+		return nil, false, err
+	}
+
+	updatedFiles := false
+	files := []string{}
+
+	lastScan := time.Now()
+	defer func() { gcd.lastScan = lastScan }()
+
+	for _, f := range globList {
+
+		info, err := os.Lstat(f)
+		if err != nil {
+			logp.Err("Error getting stats for file: %s", f)
+			continue
+		}
+
+		// Directories and symlinks are skipped
+		if !info.Mode().IsRegular() {
+			continue
+		}
```

Comment:
hm... having symlink support would be nice to have. This way op can enable modules selectively via `ln -s`... (kinda common practice in init systems, even systemd).